src/librustc_span/def_id.rs

```diff
@@ -195,6 +195,12 @@ impl fmt::Debug for DefId {
     }
 }
 
+impl From<LocalDefId> for DefId {
+    fn from(local_def_id: LocalDefId) -> DefId {
+        local_def_id.to_def_id()
+    }
+}
```

Comment:
Another reason to not use `From`/`Into` is that we kind of want people to write `.to_def_id()` instead of `.into()`... or maybe we don't.

Reply:
I am new to all of this and my opinion probably does not matter much but using `Into` is useful because we don't have to import a new trait everywhere.
Also, if i understand correctly, that new trait would only be implemented for `LocalDefId`/`DefId`. What about other key types? Using `Into` makes it more general, which may or may not be good.

Reply:
I don't think `to_def_id()` adds anything over `into`, since anecdotally most variables of type `LocalDefId` are named `def_id` anyways. We could always start with a custom trait and later migrate to `Into` if you still have concerns. We would keep an inherent method on `LocalDefId` for converting to a `DefId` so additional trait imports shouldn't be a concern.